memo-app/src/MemoForm.js

```diff
@@ -1,31 +1,48 @@
 import React from "react";
 import { useState } from "react";
+import { useAuthentication } from "./authContext.js";
 import "./MemoForm.css";
 
 export default function MemoForm({ memo, onReturnToList, onUpdate, onDelite }) {
   const [text, setText] = useState(`${memo.title}\n${memo.content}`);
   const newLine = text.indexOf("\n");
   const title = text.substring(0, newLine);
   const content = text.substring(newLine + 1);
+  const { isLoggedIn } = useAuthentication();
 
   return (
     <section id="itemB">
-      <textarea value={text} onChange={(e) => setText(e.target.value)} />
+      {isLoggedIn ? (
+        <textarea value={text} onChange={(e) => setText(e.target.value)} />
+      ) : (
+        <textarea value={text} readOnly />
+      )}
```

Comment:
こんな具合にできそう？（なるべくシンプルに、読みやすくしたいという意図）

```suggestion
<textarea
  value={text}
  onChange={(e) => setText(e.target.value)}
  readOnly={!isLoggedIn}
/>
```

Reply:
コメントありがとうございます！
`readOnly`属性に真偽値が渡せることを知りませんでした。
修正しました。